Resources/ProtobufDumper/ProtobufDumper/ProtobufDumper.cs

```diff
@@ -0,0 +1,957 @@
+        {
+            PushDescriptorName( field );
+
+            var type = ResolveType( field );
+            var options = new Dictionary<string, string>();
+
+            if ( !string.IsNullOrEmpty( field.default_value ) )
+            {
+                var defaultValue = field.default_value;
+
+                if ( field.type == FieldDescriptorProto.Type.TYPE_STRING )
+                    defaultValue = $"\"{defaultValue}\"";
+
+                options.Add( "default", defaultValue );
+            }
+            else if ( field.type == FieldDescriptorProto.Type.TYPE_ENUM && field.label != FieldDescriptorProto.Label.LABEL_REPEATED )
+            {
+                var lookup = protobufTypeMap[ field.type_name ];
+
+                if ( lookup.Source is EnumDescriptorProto enumDescriptor && enumDescriptor.value.Count > 0 )
+                    options.Add( "default", enumDescriptor.value[ 0 ].name );
+            }
+
+            var fieldOptions = DumpOptions( source, field.options );
+            foreach ( var pair in fieldOptions )
+            {
+                options[ pair.Key ] = pair.Value;
+            }
+
+            var parameters = string.Empty;
+            if ( options.Count > 0 )
+            {
+                parameters = $" [{string.Join( ", ", options.Select( kvp => $"{kvp.Key} = {kvp.Value}" ) )}]";
+            }
+
+            PopDescriptorName();
+
+            var descriptorDeclarationBuilder = new StringBuilder();
+            if ( emitFieldLabel )
+            {
+                descriptorDeclarationBuilder.Append( GetLabel( field.label ) );
+                descriptorDeclarationBuilder.Append( " " );
+            }
+
+            descriptorDeclarationBuilder.Append( $"{type} {field.name} = {field.number}{parameters};" );
+
+            return descriptorDeclarationBuilder.ToString();
+        }
+
+        private bool IsNamedType( FieldDescriptorProto.Type type )
+        {
+            return type == FieldDescriptorProto.Type.TYPE_MESSAGE || type == FieldDescriptorProto.Type.TYPE_ENUM;
+        }
+
+        public string GetPackagePath( string package, string name )
+        {
+            package = package.Length == 0 || package.StartsWith( '.' ) ? package : $".{package}";
+            return name.StartsWith( '.' ) ? name : $"{package}.{name}";
+        }
+
+        private static string GetLabel( FieldDescriptorProto.Label label )
+        {
+            switch ( label )
+            {
+                default:
+                case FieldDescriptorProto.Label.LABEL_OPTIONAL:
+                    return "optional";
+                case FieldDescriptorProto.Label.LABEL_REQUIRED:
+                    return "required";
+                case FieldDescriptorProto.Label.LABEL_REPEATED:
+                    return "repeated";
+            }
+        }
+
+        private static string GetType( FieldDescriptorProto.Type type )
+        {
+            switch ( type )
+            {
+                default:
```

Comment:
Nit: default cases belong at the bottom, unless they have the same behaviour as another case.